zfit/constraint.py

```diff
@@ -9,13 +9,14 @@
 __all__ = ["nll_gaussian", "SimpleConstraint", "GaussianConstraint"]
 
 
-def nll_gaussian(params: ztyping.ParamTypeInput, mu: ztyping.NumericalScalarType,
+def nll_gaussian(x: ztyping.NumericalScalarType, mu: ztyping.ParamTypeInput,
                  sigma: ztyping.NumericalScalarType) -> tf.Tensor:
     """Return negative log likelihood graph for gaussian constraints on a list of parameters.
 
     Args:
-        params (list(zfit.Parameter)): The parameters to constraint
-        mu (numerical, list(numerical)): The central value of the constraint
+        x (numerical, list(numerical) or list(zfit.Parameter)): Observed values of the parameter
+            to constraint obtained from auxiliary measurements
+        mu (list(zfit.Parameter)): The parameters to constraint
```

Comment:
Well RooFit the constraints are treated as gaussian pdfs that you multiply to your model pdf so I don't think RooFit is a good reference to be honest x). 
But I agree it's more intuitive, but on the other hand the observed value of the parameter is a data point so it should be in x.


So replace `x` = parameter to constraint, and `mu` is the observed value from another experiment

Reply:
Yes, it's done "wrongly" in RooFit.

It's true that it should be x _if it is a probability_. But it is not, it's a likelihood. Or: it's not a PDF but a Constraint. And in a constraint, the parametrization is different. We can also call it differently then x (although what?)

I also rather suspect that users who do not know about the subtle difference will use "x" as the parameter to constrain, right?

This also makes it "completely different" from a normal PDF (which makes also the reasoning for why this is a separate object easier :D)

Reply:
>It's true that it should be x if it is a probability. But it is not, it's a likelihood. Or: it's not a PDF but a Constraint. And in a constraint, the parametrization is different. We can also call it differently then x (although what?)

Yes it is a likelihood, and it should be an approximation of the likelihood of an auxiliary measurement, the data of the likelihood being the value that was observed in this measurement and the pdf has one free parameter.

![Capture d’écran 2020-01-30 à 15 45 32](https://user-images.githubusercontent.com/24250309/73459738-ab294800-4377-11ea-81ac-9ddcdb34698d.png)

Reply:
Yep, and L_aux takes as the first argument the parameter theta; afterwards the "parameterization". So to follow the same logic, there would also be the constraint parameter first, right?

It's different if we would build a "parametrized constraint", e.g. a PDF, and then create from that the constraint term (=likelihood).

So constraint parameter first? Or what would you prefer

Reply:
Ok I like this way of  having "parametrisation" and the the constrained parameter as you said and have indeed the constraint parameter first, like before then:
```python
GaussianConstraint(param=theta, mu=theta_obs, sigma=sigma) 
```
I have to find a way to tell `DistributionConstraint` which object should be sampled, in the case of the `GaussianConstraint` it is mu.

Reply:
> I have to find a way to tell DistributionConstraint which object should be sampled, in the case of the GaussianConstraint it is mu.
I can keep the `parameter` for that.

Reply:
> Ok I like this way of having "parametrisation" and the the constrained parameter as you said and have indeed the constraint parameter first, like before then:

well yes and no. The order stays the same, the name changes. So "mu" should maybe not be named mu, right? Rather "observed_data" (<- sure, a better name is needed)

Reply:
Ok this would be clear if we change the names of the arguments I agree. `obs` is a no go obviously, I kinda like `x` but not implicit enough maybe. Should `sigma` be renamed `uncertainty` as well?

Reply:
yeps, `obs` is taken. `observation`? 

About sigma, really no preference on that. Both seems reasonable to me. Feel free to go with whatever you prefer. If we already have `observation`, `uncertainty` would really fit nicely, if we find something better then `observation`, we may can also leave the sigma.

Reply:
I like `observation` and `uncertainty`.

Reply:
Good, let's go for it then!